objectRecognitionSource/src/main/java/edu/ml/tensorflow/util/ImageUtil.java

```diff
@@ -52,22 +60,72 @@ public void labelImage(final byte[] image, final List<Recognition> recognitions,
         }
 
         graphics.dispose();
-        saveImage(bufferedImage, "./" + outputDir + "/" + fileName);
+        saveImage(vantiq, bufferedImage, fileName);
     }
 
     /**
      * Saves an image to a location, expected to be in the directory specified in the constructor.
      * <br>Edited so that outDir will be recreated if deleted while running, vs making sure it exists initially
      * and erroring out if it disappears in the interim 
+     * @param vantiq    The VANTIQ SDK connection, either authenticated or null
      * @param image     The image to save
      * @param target    The name of the file to be written
      */
-    public void saveImage(final BufferedImage image, final String target) {
-        try {
-            IOUtil.createDirIfNotExists(new File(outputDir));
-            ImageIO.write(image,"jpg", new File(target));
-        } catch (IOException e) {
-            LOGGER.error("Unagle to save image {}!", target);
+    public void saveImage(Vantiq vantiq, final BufferedImage image, final String target) {
+        File fileToUpload = null;
+        if (outputDir != null) {
+            try {
+                IOUtil.createDirIfNotExists(new File(outputDir));
+                ImageIO.write(image,"jpg", new File(outputDir + File.separator + target));
+                fileToUpload = new File(outputDir + File.separator + target);
+            } catch (IOException e) {
+                LOGGER.error("Unable to save image {}", target, e);
+            }
+        }
+        if (vantiq != null) {
+            if (fileToUpload != null) {
+                vantiq.upload(fileToUpload, 
```

Comment:
Seems like this should all become
```
if (fileToUpload == null) {
   fileToUpload = File.createTempFile()
}

vantiq.upload(fileToUpload...)
```
No real reason to make have that call twice.

Reply:
There are two reasons why I repeat the vantiq.upload twice, let me know what you think.

1. One version of the vantiq.upload deletes the file onSuccess, but the other should not do this. I can move the deleting file part somewhere else which would fix this issue, but I wasn't sure if we'd want to delete the image if it wasn't successfully sent.

2. Since fileToUpload is a local variable, when I tried to use that variable and call the "fileToUpload.delete()" in onSuccess, I kept getting this error:
    * "Local variable fileToUpload defined in an enclosing scope must be final or effectively final"

Let me know what you think I should do here.